dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/RelationshipsValidationHook.java

```diff
@@ -189,6 +209,26 @@ else if ( constraint.getRelationshipEntity().equals( PROGRAM_STAGE_INSTANCE ) )
         return result;
     }
 
+    private Optional<String> getRelationshipTypeUidFromTrackedEntity( TrackerImportValidationContext ctx, String uid )
+    {
+        return getTrackedEntityTypeFromTrackedEntity( ctx, uid ).map( Optional::of )
```

Comment:
is this .map(Optional::of) needed?

Reply:
Yes, without it the code would not compile. See image

<img width="803" alt="Screen Shot 2020-11-24 at 10 56 03" src="https://user-images.githubusercontent.com/300784/100078368-bcca6780-2e43-11eb-9006-b6cfc0f247c6.png">